scripts/cert-manager/app-registration-cert-mananger.sh

```diff
@@ -0,0 +1,308 @@
+            printf " ERROR: Could not add user \"$USER_PRINCIPAL_NAME\" as owner of app registration.\n"
+        fi
+
+    done
+    printf " Done.\n"
+fi
+
+#######################################################################################
+### Refresh credential secret
+###
+CRED_SECRETS=true
+echo ""
+if [[ $USER_PROMPT == true ]]; then
+    while true; do
+        read -p "Refresh credential secret? (Y/n) " yn
+        case $yn in
+            [Yy]* ) break;;
+            [Nn]* ) CRED_SECRETS=false; echo "Skipping."; break;;
+            * ) echo "Please answer yes or no.";;
+        esac
+    done
+fi
+
+if [[ $CRED_SECRETS == true ]]; then
+    printf "Refresh credential secret..."
+    APP_DESCRIPTION="$APP_REGISTRATION_CERT_MANAGER"
+    UPDATED_CLIENT_SECRET=$(az ad app credential reset --id "$APP_ID" --credential-description "$APP_DESCRIPTION" 2>/dev/null) # For some reason, description can not be too long.
+
+    if [[ $UPDATED_CLIENT_SECRET == "" ]]; then
+        printf " ERROR: Could not refresh client secret for App Registration \"$APP_REGISTRATION_CERT_MANAGER\". Exiting...\n"
+        exit 1
+    fi
+    printf " Done.\n"
+
+    APP_PASSWORD=$(echo $UPDATED_CLIENT_SECRET | jq '.password')
+    APP_NAME=$(echo $UPDATED_CLIENT_SECRET | jq '.name')
+    APP_TENANT=$(echo $UPDATED_CLIENT_SECRET | jq '.tenant')
+
+
+    # Get expiration date of updated credential
+    EXPIRATION_DATE=$(az ad app credential list --id $APP_ID --query "[?customKeyIdentifier=='$APP_DESCRIPTION'].endDate" --output tsv | sed 's/\..*//')"Z"
+
+    # Create new .json file with updated credential.
+    UPDATED_SECRET_VALUES_FILE="updated_secret_values.json"
+    test -f "$UPDATED_SECRET_VALUES_FILE" && rm "$UPDATED_SECRET_VALUES_FILE"
+
+    cat <<EOF >>${UPDATED_SECRET_VALUES_FILE}
+{
+"name": "${APP_REGISTRATION_CERT_MANAGER}",
+"id": "${APP_ID}",
+"password": "${APP_PASSWORD}",
+"description": "${APP_DESCRIPTION}",
+"tenantId": "${APP_TENANT}"
+}
+EOF
+
+    # Update keyvault with secret
+    printf "Updating keyvault \"$AZ_RESOURCE_KEYVAULT\"..."
+    if [[ $(az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE" 2>&1) == *"ERROR"* ]]; then
+        az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE"
+        echo -e "\nERROR: Could not update secret in keyvault \"$AZ_RESOURCE_KEYVAULT\". Exiting..."
+        exit 1
+    fi
+    printf " Done\n"
+
+    rm $UPDATED_SECRET_VALUES_FILE
+fi
+
+#######################################################################################
+### Assign custom permission on TXT records
+###
+
+ROLENAME="DNS TXT Contributor"
+APP_ID=$(az role definition list --name "$ROLENAME" -o tsv)
+CRED_ROLE=true
+echo ""
+if [[ $APP_ID == "" ]]; then
+    echo "Role \"$ROLENAME\" does not exist."
```

Comment:
Specify that it is a role definition
```suggestion
    echo "Role definition \"$ROLENAME\" does not exist."
```